lua/pac3/editor/server/wear.lua

```diff
@@ -108,68 +122,66 @@ duplicator.RegisterEntityModifier("pac_config", function(ply, ent, parts)
 	end)
 end)
 
-function pace.SubmitPart(data, filter)
-	if istable(data.part) then
-		if last_frame == frame_number then
-			table.insert(pace.StreamQueue, {data, filter})
-			pace.dprint("queuing part %q from %s", data.part.self.Name, tostring(data.owner))
-			return "queue"
-		end
+function pace.SubmitPartNow(data, filter)
+	local part = data.part
+	local owner = data.owner
+
+	-- last arg "true" is pac3 only in case you need to do your checking differently from pac2
+	local allowed, reason = hook.Run("PrePACConfigApply", owner, data, true)
+	if allowed == false then return allowed, reason end
+
+	local uid = data.uid
+	if uid ~= false and pace.IsBanned(owner) then
+		return false, "you are banned from using pac"
 	end
```

Comment:
I moved this banned check up farther too, seems like something we'd want to return early for.